File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/History/HistoryDate.cs
namespace Psi.Data.Models.ClientConfigurationModels.History
{
    public class HistoryDate : SettingsBaseHelper
    {
        public HistoryDate(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("History.Date.DateRangeDropDownSettingOptions")]
        public string DateRangeDropDownSettingOptions
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("History.Date.DateRangeDropDownSettingLabelValue")]
        public string DateRangeDropDownSettingLabelValue
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("History.Date.DatePickerPositionOptions")]
        public string DatePickerPositionOptions
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }
    }
}

File: infrastructure/models/ClientConfigurationModels/History/HistoryDate.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
export interface HistoryDateConfig {
    DateRangeDropDownSettingOptions: string;
    DateRangeDropDownSettingLabelValue: string;
    DatePickerPositionOptions: string;
}

export class HistoryDate implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'HistoryDate'
    };


            private _dateRangeDropDownSettingOptions: string;
            get dateRangeDropDownSettingOptions(): string {
                return this._dateRangeDropDownSettingOptions;
            }
            set dateRangeDropDownSettingOptions(value: string) {
                this._dateRangeDropDownSettingOptions = value;
            }

            private _dateRangeDropDownSettingLabelValue: string;
            get dateRangeDropDownSettingLabelValue(): string {
                return this._dateRangeDropDownSettingLabelValue;
            }
            set dateRangeDropDownSettingLabelValue(value: string) {
                this._dateRangeDropDownSettingLabelValue = value;
            }

            private _datePickerPositionOptions: string;
            get datePickerPositionOptions(): string {
                return this._datePickerPositionOptions;
            }
            set datePickerPositionOptions(value: string) {
                this._datePickerPositionOptions = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "HistoryDate.DateRangeDropDownSettingOptions", value: this._dateRangeDropDownSettingOptions, dataType: 'string', label: "Date Range Drop Down Setting Options" },
                { key: "HistoryDate.DateRangeDropDownSettingLabelValue", value: this._dateRangeDropDownSettingLabelValue, dataType: 'string', label: "Date Range Drop Down Setting Label Value" },
                { key: "HistoryDate.DatePickerPositionOptions", value: this._datePickerPositionOptions, dataType: 'string', label: "Date Picker Position Options" },
            ];
        }

}